spec_opts.go

```diff
@@ -29,16 +30,16 @@ func WithHostname(name string) SpecOpts {
 
 // WithNewSpec generates a new spec for a new container
 func WithNewSpec(opts ...SpecOpts) NewContainerOpts {
+	return WithNewPlatformSpec(platforms.Default().OS, opts...)
+}
+
+// WithNewPlatformSpec generates a new spec for a new container
```

Comment:
This combinator doesn't read correctly: this would mean that you are adding a platform spec to the container. Can this not be a spec opt?

Reply:
The reason this is not a spec opt is that, in the current code, spec opts are applied to an already generated default spec, but this needs to change the generated spec used, as LCOW use essentially a Linux spec with a little bit of Windows config on top, rather than the other way around. I'll see if I can find a way to cleanly make this a spec opt.

Reply:
Yeah, I don't think this works as a spec opt currently. It needs to overwrite so much of the Spec that, depending on the order the specs are applied, other spec opts might be overwritten.

One example is Root.Process which has different defaults, as well as common opts that affect those same defaults. Cwd, the default on Linux is "/" the default on Windows is "C:\", and it would seem common to use a spec opt to change this. If the platform selection is used as a spec opt, then the order of application of the default value could overwrite a specified value. I think either the withPlatform spec opt would need to be carefully written to avoid overwriting, or it should have some sort of priority over other spec opts so it runs first. While I think both of these are possible, neither feels clean. It seems like what I really want to do here is change the starting spec, hence `WithNewPlatformSpec(...)`.

Essentially we need to know which default to apply the spec opts too, so setting that default as a spec opt feels weird to me.